### ISSUE TITLE
Request: QOI (Quite OK Image) Format

### ISSUE THREAD

**Issue by John-P** [cgohlke/imagecodecs#37]
Just a suggestion of an additional codec to include. The QOI (Quite Okay Image) format is a single MIT C/C++ binary available here https://github.com/phoboslab/qoi. It has significantly faster encode/decode than PNG for a similar compression ratio. There is [already python package](https://github.com/kodonnell/qoi), but it might make a nice addition to imagecodecs.

**Comment by tasansal** [comment]
+1 for this one.

**Comment by barbara73** [comment]
Hey
I have a question. 
I used your tool for jpeg2000 lossless compression:

`compressed_img = jpeg2k_encode(ds.pixel_array, level=0.0, reversible=1)  #2`

One can immediately decompress the image with decode. Everything fine.
But after

```
from pydicom.encaps import encapsulate
from pydicom.uid import JPEG2000Lossless

ds.is_little_endian = True
ds.is_implicit_VR = False
ds.file_meta.TransferSyntaxUID = JPEG2000Lossless
ds.PhotometricInterpretation = 'RGB'
ds.PixelData = encapsulate([compressed_img])  #1
ds['PixelData'].is_undefined_length = True

ds.save_as("temp_test.dcm", write_like_original=False)
```

But after encapsulating the pixel data with pydicom (#1), I am not able to decompress with your tool.
Can you tell me why? How can I be sure, that the image is indeed reversible?
I found out, that the binary data (pixel data after encapsulating - see #1) has an additional byte-string in front of the compressed_image #2.

Am I missing something?
Thank you
Barbara

**Comment by cgohlke** [comment]
A QOI codec is included in v2022.7.27.
